src/main/java/org/generationcp/middleware/dao/germplasmlist/GermplasmListDAO.java

```diff
@@ -883,4 +903,118 @@ public long countSearchGermplasmList(final GermplasmListSearchRequest germplasmL
 
 		return ((BigInteger) query.uniqueResult()).longValue();
 	}
+
+	public long countGermplasmListDTOs(final GermplasmListSearchRequestDTO requestDTO) {
+		final SQLQuery sqlQuery = this.getSession().createSQLQuery(this.createCountListsQueryString(requestDTO));
+		this.addListSearchParameters(sqlQuery, requestDTO);
+		return ((BigInteger) sqlQuery.uniqueResult()).longValue();
+	}
+
+	public List<GermplasmListDTO> searchGermplasmListDTOs(final GermplasmListSearchRequestDTO requestDTO, final Pageable pageable) {
+		final SQLQuery sqlQuery = this.getSession().createSQLQuery(this.createListsQuery(requestDTO));
+		if (pageable != null) {
```

Comment:
This code is duplicated:
`		if (pageable != null) {
			sqlQuery.setFirstResult(pageable.getPageSize() * pageable.getPageNumber());
			sqlQuery.setMaxResults(pageable.getPageSize());
		}`

Reply:
You can use `addPaginationToSQLQuery` instead